scripts/useReactVersion.mjs

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-console */
+/**
+ * Given the dist tag fetch the corresponding
+ * version and make sure this version is used throughout the repository.
+ *
+ * If you work on this file:
+ * WARNING: This script can only use built-in modules since it has to run before
+ * `pnpm install`
+ */
+import childProcess from 'child_process';
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { promisify } from 'util';
+import { getWorkspaceRoot } from './utils.mjs';
+
+// TODO: reuse the `useReactVersion.mjs` from the monorepo
+
+const exec = promisify(childProcess.exec);
+
+// packages published from the react monorepo using the same version
+const reactPackageNames = ['react', 'react-dom', 'react-is', 'react-test-renderer', 'scheduler'];
+const devDependenciesPackageNames = ['@mnajdova/enzyme-adapter-react-18', '@testing-library/react'];
+
+// if we need to support more versions we will need to add new mapping here
+const additionalVersionsMappings = {
+  17: {
+    '@mnajdova/enzyme-adapter-react-18': 'npm:@eps1lon/enzyme-adapter-react-17',
+    '@testing-library/react': '^12.1.0',
+  },
+  19: {},
```

Comment:
Is it only this entry that warranted the need to duplicate this file instead of using the one from `monorepo`? 🤔 
Is it necessary though? 🤔

Reply:
That's one of the reasons.
Another one was that I wasn't sure if the workspace root would be resolved correctly if using the file from the monorepo: https://github.com/mui/material-ui/blob/f993fe56f59d80c9b255c0585b6b428689dfea18/scripts/useReactVersion.mjs#L15
I'll give it a try

Reply:
> Another one was that I wasn't sure if the workspace root would be resolved correctly if using the file from the monorepo: https://github.com/mui/material-ui/blob/f993fe56f59d80c9b255c0585b6b428689dfea18/scripts/useReactVersion.mjs#L15

That script will definitely return the monorepo root.
That was the main reason why the `yarn deduplicate` script didn't work on `mui-x`. 🙈 
https://github.com/mui/material-ui/blob/v5.13.5/scripts/deduplicate.mjs